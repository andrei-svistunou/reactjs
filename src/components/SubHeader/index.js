import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../Logo';
import './index.css';

const SubHeader = props => (
  <div className="flex">
    <Logo />
    {props.children}
  </div>
);

SubHeader.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.element), PropTypes.element])
};

export default SubHeader;
