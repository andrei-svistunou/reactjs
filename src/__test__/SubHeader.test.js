import React from 'react';
import { shallow } from 'enzyme';
import SubHeader from '../components/SubHeader';

describe('<SubHeader />', () => {
  describe('Shapshot', () => {
    test('should render <SubHeader /> without children', () => {
      const wrapper = shallow(<SubHeader />);

      expect(wrapper).toMatchSnapshot();
    });

    test('should render <SubHeader /> with children', () => {
      const wrapper = shallow(<SubHeader><p>test children</p></SubHeader>);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
