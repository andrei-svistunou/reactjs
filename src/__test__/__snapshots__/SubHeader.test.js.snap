// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubHeader /> Shapshot should render <SubHeader /> with children 1`] = `
<div
  className="flex"
>
  <Logo
    text="netflixroulett"
  />
  <p>
    test children
  </p>
</div>
`;

exports[`<SubHeader /> Shapshot should render <SubHeader /> without children 1`] = `
<div
  className="flex"
>
  <Logo
    text="netflixroulett"
  />
</div>
`;
